twill-core/src/main/java/org/apache/twill/kafka/client/KafkaConsumer.java

```diff
@@ -33,14 +33,16 @@
 
     /**
      * Invoked when new messages is available.
+     * @param startOffset Offset of the first {@link FetchedMessage} in the iterator of new messages.
```

Comment:
The description yes, but I think you need to change the implementation to honor it.